WcaOnRails/app/models/user.rb

```diff
@@ -633,16 +633,19 @@ def serializable_hash(_options = nil)
       },
     }
 
-    if doorkeeper_token
+    include_email = self.any_kind_of_delegate? # Delegates's emails are public information.
+    if doorkeeper_token && doorkeeper_token.resource_owner_id == self.id
```

Comment:
This is a tangentially related change: we only set `doorkeeper_token` for the `/me` route, so it was safe to not check if it corresponds to the user, but it's definitely something we should be checking, IMO.